dandi/tests/test_models.py

```diff
@@ -154,3 +166,265 @@ def test_autogenerated_titles():
+    of requests.post and requests.get
+    """
+
+    # removing doi in case it exists
+    _clean_doi(doi)
+
+    # checking f I'm able to create doi
+    rp = requests.post(
+        "https://api.test.datacite.org/dois",
+        json=datacite,
+        headers={"Content-Type": "application/vnd.api+json"},
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+    # checking if i'm able to get the url
+    rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
+
+    # cleaning url
+    _clean_doi(doi)
+    return rp.status_code, rg.status_code
+
+
+def _clean_doi(doi):
+    """removing doi, ignoring the status code"""
+    requests.delete(
+        f"https://api.test.datacite.org/dois/{doi}",
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+
+@pytest.fixture(scope="module")
+def schema():
+    sr = requests.get(
+        "https://raw.githubusercontent.com/datacite/schema/master/source/"
+        "json/kernel-4.3/datacite_4.3_schema.json"
+    )
+    schema = sr.json()
+    return schema
+
+
+@pytest.fixture(scope="module")
+def client():
+    return DandiAPIClient("https://api.dandiarchive.org/api/")
+
+
+@pytest.mark.parametrize("dandi_id", ["000004", "000008"])
+def test_datacite(dandi_id, schema, client):
+    """ checking to_datacite for a specific datasets"""
+    prefix = "10.80507"
+    version = "draft"
+
+    # getting metadata from the dandiarchive api
+    newmeta_js = client.get_dandiset(dandi_id, version)["metadata"]
+
+    newmeta_js["doi"] = f"{prefix}/dandi.{random.randrange(99)}{dandi_id}.{version}"
+    newmeta_js["datePublished"] = str(datetime.now().year)
+    newmeta_js["publishedBy"] = "https://doi.test.datacite.org/dois"
+    newmeta_js["version"] = version
+    newmeta = PublishedDandisetMeta(**newmeta_js)
+
+    datacite = to_datacite(meta=newmeta)
+
+    Draft6Validator.check_schema(schema)
+    validator = Draft6Validator(schema)
+    validator.validate(datacite["data"]["attributes"])
+
+    # posting datacite, and checking status codes
+    post_status_code, get_status_code = datacite_post(datacite, newmeta.doi)
+    assert post_status_code == 201
+    assert get_status_code == 200
+
+
+def test_dantimeta_1():
+    """ checking basic metadata for publishing"""
+    # meta data without doi, datePublished and publishedBy
+    meta_dict = {
+        "identifier": "DANDI:912",
+        "id": "DANDI:912/draft",
+        "version": "v.1",
+        "name": "testing dataset",
+        "description": "testing",
+        "contributor": [
+            {
+                "name": "last name, first name",
+                "roleName": [RoleType("dandi:ContactPerson")],
+            }
+        ],
+        "license": [LicenseType("spdx:CC-BY-4.0")],
+    }
+
+    # should work for DandisetMeta but PublishedDandisetMeta should raise an error
+    DandisetMeta(**meta_dict)
+    with pytest.raises(Exception) as exc:
```

Comment:
Below, you seem to be assuming that the exception is a pydantic `ValidationError`.  This should thus be changed to `with pytest.raises(ValidationError) as exc:`.